### PR TITLE
CodeSysytemName -> Codesystemoid

### PR DESCRIPTION
**Pull request by mayerm94**
Use oid in codes instead of name for code system
Pull requests into cqm-converter require the following. Submitter and reviewer should :white_check_mark: when done. For items that are not-applicable, note it's not-applicable ("N/A") and :white_check_mark:.

❗️BEFORE MERGE ❗️
- [x] AFTER https://github.com/projecttacoma/cqm-models/tree/codesystemoid update gemfiles etc.
**Submitter:**

- [x] This pull request describes why these changes were made.
- [x] Internal ticket for this PR: https://jira.mitre.org/browse/CYPRESS-519
- [x] Internal ticket links to this PR
- [x] Code diff has been done and been reviewed
- [x] Tests are included and test edge cases
- [x] Tests have been run locally and pass
- [x] If applicable, the library version number in `cqm-converter.gemspec` has been updated NA
- [x] Update cqm-execution test fixtures using update_cqm_execution_fixtures.sh script from server-scripts repo

**Bonnie Reviewer:**

Name:
- [x] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [x] The tests appropriately test the new code, including edge cases
- [x] You have tried to break the code

**Cypress Reviewer:**

Name:
- [ ] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [ ] The tests appropriately test the new code, including edge cases
- [ ] You have tried to break the code

### CODE REVIEW

File: spec/cqm/converter/bonnie_measure_spec.rb [hunk 1]
```diff
@@ -1,5 +1,5 @@
 require 'spec_helper'
-
+require 'pry'
```
Comment (jbradl11) [inline spec/cqm/converter/bonnie_measure_spec.rb]: I don't think you want this here.

Comment (mayerm94) [reply spec/cqm/converter/bonnie_measure_spec.rb]: I don't think I do

File: lib/cqm/converter/bonnie_measure.rb [hunk 2]
```diff
@@ -129,6 +131,19 @@ def self.measure_and_valuesets_to_cqm(bonnie_measure, hds_valuesets)
       cqm_measure.value_sets = cqm_valuesets
       cqm_measure
     end
+
+    def self.update_oids(cql_library)
+      name_oid_hash = JSON.parse(File.read(File.join(File.dirname(__FILE__), 'name_oid_map.json')))
+      return unless cql_library['elm']['library'] && cql_library['elm']['library']['codeSystems'] && cql_library['elm']['library']['codeSystems']['def']
+
+      cql_library['elm']['library']['codeSystems']['def'].each do |codesystem|
```
Comment (jbradl11) [inline lib/cqm/converter/bonnie_measure.rb]: *You do not need to use this recommendation* but a slightly cleaner (more ruby solution) would be 
```
(cql_library.dig('elm', 'library', 'codeSystems', 'def') || []).each do |codesystem|
```
this will allow you to remove the return clause on line 137 because dig is a safe way to "dig" through a hash.

File: lib/cqm/converter/hds_record.rb [hunk 3]
```diff
@@ -163,8 +158,33 @@ def generate_qdm_data_element(dc_fixed_keys, dc_type)
           QDM::FacilityLocation.new.from_json(facility.to_json)
         end
       end
-
+      populate_codesystem_oid(data_element)
+      # iterate over all of the fields
+      # if array or one of the nested types (facility location) dive deeper
+      # if object contains codeSystem and codeSystemOid doesn't have a value then set codeSystemOid to the correct OID
```
Comment (jbradl11) [inline lib/cqm/converter/hds_record.rb]: I think these comments should be right above the `populate_codesystem_oid` function definition.

File: lib/cqm/converter/hds_record.rb [hunk 4]
```diff
@@ -163,8 +158,33 @@ def generate_qdm_data_element(dc_fixed_keys, dc_type)
...
       data_element
     end
+
+    def populate_codesystem_oid(entry)
```
Comment (jbradl11) [inline lib/cqm/converter/hds_record.rb]: i have a few questions about this function. Let's talk in the morning.

File: lib/cqm/converter/hds_record.rb [hunk 5]
```diff
@@ -163,8 +158,33 @@ def generate_qdm_data_element(dc_fixed_keys, dc_type)
...
+      if entry.nil? || entry.is_a?(String) || entry.is_a?(BSON::ObjectId) ||
+         entry.is_a?(Time) || entry.is_a?(Date) || entry.is_a?(Boolean) || entry.is_a?(Integer) || entry.is_a?(Float)
+      elsif entry.is_a?(Array)
+        entry.each { |elem| populate_codesystem_oid(elem) }
+      elsif entry.is_a?(Hash)
+        if entry['codeSystem'] || entry[:codeSystem]
+          name_oid_hash = JSON.parse(File.read(File.join(File.dirname(__FILE__), 'name_oid_map.json')))
+          if name_oid_hash[entry['codeSystem']].nil? && name_oid_hash[entry[:codeSystem]].nil?
+            puts 'ERROR: Could Not Resolve OID For Code System ' + entry['codeSystem']
```
Comment (jbradl11) [inline lib/cqm/converter/hds_record.rb]: I see this error popup when running the converter for the following codeSystems: 
- "ICD-9-PCS" 
- "rxnorm"
- "ICD 10"
- "MyCode"
- "mycode"

Now, clearly atleast 2 of these are not real code systems. But we need to discuss what happens if they aren't converted over / what accounts they are on.

Comment (mayerm94) [reply lib/cqm/converter/hds_record.rb]: Addes support for "ICD-9-PCS", "rxnorm", "ICD 10" names. Will have to discuss what to do about "myCode"
